src/app_state.rs

```diff
@@ -0,0 +1,305 @@
+use crate::app_dirs::AppDirs;
+use crate::config::Config;
+use crate::process::SyncthingProcess;
+
+/// Syncthingers application state.
+pub struct AppState {
+    pub config: Config,
+    pub syncthing_process: Option<SyncthingProcess>,
+    pub app_dirs: AppDirs,
+}
+
+impl AppState {
+    pub fn new(config: Config, app_dirs: AppDirs) -> Self {
+        // Skip process detection for test environments
+        let syncthing_process = if config.syncthing_path.contains("test")
```

Comment:
This if sentence is repeated multiple times in several functions. It would be nice to have it as an own function. It could also use a list of `["test", "mock", "nonexistent"]` that can be checked against `config.syncthing_path`.